### DISCUSSION TITLE
Sort by multiple fields, and configurable null sorting

### DISCUSSION THREAD

**Discussion by tv42** [keystonejs/keystone#7422] [Feature requests]
I have a hierarchy of categories managed in KeystoneJS. For the sake of conversation, think of eCommerce categories like "Pet supplies > Fish > Fish food > Tropical fish food".

In the list for those categories, sorting by just one column leads to not-great results, the order makes little sense to a human. Currently, at best, I can do something like

```
        initialSort: { field: "parent", direction: "ASC" },
```

To at least group items by their immediate parent, but the larger tree shape is all jumbled.

What I'd like to be able to do is something like

```
initialSort: [
  // First, sort by parent relationship's field `displayOrder`.
  // Null parent means top level of tree, place those first.
  { field: "parent.displayOrder", direction: "ASC", nullPlacement: "first" },
  // Then, sort by display order within that parent.
  // Items with `displayOrder` unset go last.
  { field: "displayOrder", direction: "ASC", nullPlacement: "last" },
  // Then alphabetically by name
  { field: "name", direction: "ASC" },
  // Resolve ties by id
  { field: "id", direction: "ASC" },
],
```

Any thoughts?

**Comment by asgeo1** [comment]
Even if it just had ability to multi-sort on columns present in the model itself, would be far more useful than just sorting by a single column
